zerver/decorator.py

```diff
@@ -417,39 +458,46 @@ def _wrapped_view_func(request, *args, **kwargs):
 
 # Converter functions for use with has_request_variables
 def to_non_negative_int(x):
+    # type: (float) -> int
     x = int(x)
     if x < 0:
         raise ValueError("argument is negative")
     return x
 
 def to_non_negative_float(x):
+    # type: (float) -> float
     x = float(x)
     if x < 0:
         raise ValueError("argument is negative")
     return x
 
 def flexible_boolean(boolean):
+    # type: (text_type) -> bool
     """Returns True for any of "1", "true", or "True".  Returns False otherwise."""
     if boolean in ("1", "true", "True"):
         return True
     else:
         return False
 
 def statsd_increment(counter, val=1):
+    # type: (text_type, int) -> Callable[[Callable[..., Any]], Callable[..., Any]]
     """Increments a statsd counter on completion of the
     decorated function.
 
     Pass the name of the counter to this decorator-returning function."""
     def wrapper(func):
+        # type: (Callable[..., Any]) -> Callable[..., Any]
         @wraps(func)
         def wrapped_func(*args, **kwargs):
+            # type: (*Any, **Any) -> Any
             ret = func(*args, **kwargs)
             statsd.incr(counter, val)
             return ret
         return wrapped_func
     return wrapper
 
 def rate_limit_user(request, user, domain):
+    # type: (HttpRequest, UserProfile, str) -> None
```

Comment:
I think domain should be `text_type`. I fetched your pull request and tried changing `str` to `text_type`. I got a few mypy errors. I tried to trace the cause and I think that annotations in `zerver/lib/rate_limiter.py` are incorrect. I replaced all `str`s in it by `text_type`. I didn't get any errors after that.

Can you fix `zerver/lib/rate_limiter.py` in this pull request? A simple find-and-replace would do the job. Then put that in a commit of its own.
FYI @timabbott, @ashishk1994.